charmhelpers/contrib/network/ovs/utils.py

```diff
@@ -0,0 +1,27 @@
+# Copyright 2019 Canonical Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import subprocess
+
+
+def _run(*args):
+    """Run a process, check result, capture decoded output from STDERR/STDOUT.
+
+    :param args: Command and arguments to run
+    :type args: Tuple[str, ...]
+    :returns: Information about the completed process
+    :rtype: str
+    :raises subprocess.CalledProcessError
+    """
+    return subprocess.check_output(args, stderr=subprocess.STDOUT,
+                                   universal_newlines=True)
```

Comment:
Bearing in mine the recent discussion regarding piping STDERR to STDOUT and the brittleness that this provides, should it be captured to STDOUT?  i.e. is it useful to capture the STDERR as part of the output of the function?

Reply:
Reviewing the usage of this utility function I see no need for capturing ``STDERR`` at this time, I'll drop it and we can add handling of that as needed.

The original version of the code used the modern ``subprocess.run`` interface which give you a separated view of ``STDOUT`` and ``STDERR``in the ``CompletedProcess`` response, but sadly we cannot use such modern interfaces for code in charm-helpers, so it got lost in translation.